R/arules.R

```diff
@@ -140,14 +140,20 @@ do_apriori <- function(df, subject, key, minlen=1, maxlen=10, min_support=0.1, m
   do_apriori_(df, subject_col, key_col, minlen, maxlen, min_support, max_support, min_confidence, lhs, rhs)
 }
 
-get_arules_graph_data <- function(rules) {
-  rules <- rules %>% dplyr::top_n(30, support) # limit within 30 rules so that they can be visualized comfortably.
-  if (nrow(rules) > 30) { # this means there are ties. remove the rows with minimum support to fit within 30 rules.
-    rules <- rules %>% dplyr::filter(support != min(support))
+# rules_metric can be "support", "confidence", or "lift".
+get_arules_graph_data <- function(rules, max_rules=30, rules_metric="support") {
+  rules <- rules %>% dplyr::top_n(max_rules, UQ(rlang::sym(rules_metric))) # limit within 30 rules so that they can be visualized comfortably.
+  if (nrow(rules) > max_rules) { # this means there are ties. remove the rows with minimum support to fit within 30 rules.
+    if (!(rules_metric == "confidence" && min(rules$confidence) == 1)) { # exception is when supports for all rules are 1.0.
+      rules <- rules %>% dplyr::filter(UQ(rlang::sym(rules_metric)) != min(UQ(rlang::sym(rules_metric))))
+    }
+    else {
+      rules <- rules %>% sample_n(max_rules) # in this case, just sample so that plotting will not take very long time.
+    }
   }
 
   # Give names to the rules. groceries is the dataframe that is the result of the Market Basket Analysis.
-  rules <- rules %>% dplyr::mutate(rule = row_number(), rule = str_c("Rule ",parse_character(rule)))
+  rules <- rules %>% mutate(rule = row_number(), rule = stringr::str_c("Rule ",readr::parse_character(rule)))
```

Comment:
Why you explicitly took off dplyr:: here?

Reply:
I think I mistakenly did this when I had to remove dplyr:: from row_number to pass test on linux. Put it back.